app/src/main/java/com/github/muellerma/prepaidbalance/ui/PreferenceActivity.kt

```diff
@@ -68,6 +73,22 @@ class PreferenceActivity : AppCompatActivity() {
                 }
             }
 
+            if (ActivityCompat.checkSelfPermission(requireContext(),
+                    Manifest.permission.READ_PHONE_STATE
+                ) == PackageManager.PERMISSION_GRANTED) {
```

Comment:
If not granted, ask for permissions and disable setting.

Reply:
> If not granted, ask for permissions and disable setting.

@mueller-ma ,
Made this change. Please update the review